data_creation/data_creation.py

```diff
@@ -0,0 +1,357 @@
+        Loads the DGS Types dataset and creates a gloss-to-pose dictionary.
+        """
+        config = SignDatasetConfig(
+            name="pose_holistic",
+            version="3.0.0",
+            include_video=False,
+            include_pose="holistic",
+            process_pose=False
+        )
+        dgs_types = tfds.load('dgs_types', 
+                              builder_kwargs=dict(config=config), 
+                              data_dir=self.dictionary_dir)
+        dgs_types_dict, _ = create_gloss_to_pose_dict(dgs_types)
+        return dgs_types_dict
+
+    def _should_replace(self, gloss):
+        """
+        Determines whether a gloss's pose should be replaced.
+        """
+        # Check if gloss is in dgs types dictionary and has a pose, and is not a special generic sign
+        return not gloss.startswith('$') and gloss in self.dictionary and self.dictionary[gloss]['views']['pose'] is not None
+
+    @lru_cache(maxsize=None)
+    def _get_pose_header(self, dataset_name: str):
+        """
+        Retrieves the pose header for a given dataset.
+        :param dataset_name: Name of the dataset (e.g., "dgs_corpus").
+        :return: PoseHeader object.
+        """
+        try:
+            # Check cache first
+            if dataset_name in self._pose_headers_cache:
+                return self._pose_headers_cache[dataset_name]
+
+            # Dynamically import the dataset module
+            dataset_module = importlib.import_module(f"sign_language_datasets.datasets.{dataset_name}.{dataset_name}")
+            
+            # Read the pose header from the dataset's predefined file
+            with open(dataset_module._POSE_HEADERS["holistic"], "rb") as buffer:
+                pose_header = PoseHeader.read(BufferReader(buffer.read()))
+            
+            # Cache the pose header
+            self._pose_headers_cache[dataset_name] = pose_header
+            return pose_header
+        except Exception as e:
+            print(f"Error getting pose header for {dataset_name}: {e}")
+            raise
+
+    def _create_pose_object(self, pose_datum, dataset_name: str):
+        """
+        Creates a Pose object from the given pose data_entry and dataset name.
+        :param pose_datum: Pose data_entry dictionary containing fps, data_entry, and conf.
+        :param dataset_name: Name of the dataset.
+        :return: Pose object.
+        """
+        try:
+            # Retrieve the pose header for the dataset
+            pose_header = self._get_pose_header(dataset_name)
+            
+            # Extract pose data_entry
+            fps = float(pose_datum["fps"].numpy())
+            pose_data = pose_datum["data"].numpy()
+            conf = pose_datum["conf"].numpy()
+            pose_body = NumPyPoseBody(fps, pose_data, conf)
+            
+            # Construct and return the Pose object
+            pose = Pose(pose_header, pose_body)
+            return pose
+        except Exception as e:
```

Comment:
does this except ever happen? even if so, it will be communicative enough, no need to catch and rethrow, i think